code/modules/mod/mod_control.dm

```diff
@@ -142,6 +142,7 @@
 		install(module)
 	RegisterSignal(src, COMSIG_ATOM_EXITED, .proc/on_exit)
 	RegisterSignal(src, COMSIG_SPEED_POTION_APPLIED, .proc/on_potion)
+	RegisterSignal(src, COMSIG_ITEM_SUMMONED, .proc/on_summon)
```

Comment:
i think itd be better to handle it on Move() so it handles all edge cases

Reply:
what do you mean by this